### PR TITLE
feat:Adding support for KubeArmorClusterPolicy

### LINKED ISSUE #1769: Cluster Scoped KubeArmor Security Policy
## Feature Request

**Short Description**

As a User, I want to apply a policy which is applicable across cluster where I should be able to mention exclusions. 

**Is your feature request related to a problem? Please describe the use case.**

Need to write policies for each namespace manually

**Describe the solution you'd like**

Create a new CRD with ClusterKubeArmorSecurityPolicy. Example Policy:

```diff
apiVersion: security.kubearmor.com/v1
+kind: KubeArmorClusterPolicy
metadata:
  name: ksp-block-cis-tmp-no-exec
spec:
  tags: ["CIS"]
  message: "Alert! exec tmp!"
  selector:
+    exclude:
+         namespace:
+          - kube-system
+          - kubearmor
  process:
    severity: 1
    matchDirectories:
    - dir: /tmp/
      recursive: true
    action: Block
```

**Describe alternatives you've considered**
Manually create for each namespace

### PR DESCRIPTION
**Pull request by Prateeknandle**
**Purpose of PR?**:
Adding support for KubeArmorClusterPolicy

Feature #1769

**Does this PR introduce a breaking change?**

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_
1. Added a separate watcher `WatchClusterSecurityPolicies()` for `KubeArmorClusterPolicy`.
2. Leverage `CreateSecurityPolicy()` for creating security policy object:
    - To handle `matchExpressions`, added `NamespaceList` to `tp.SecurityPolicy`.
    - `NamespaceList` contains list of namespaces on which cluster policy will be enforced.
    - we do all computations w.r.t. `matchExpressions` in `CreateSecurityPolicy()`.
3. `UpdateSecurityPolicy()` is used to update `endPoints` security policies.

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [x] Commit has integration tests

### CODE REVIEW

File: KubeArmor/core/kubeUpdate.go [hunk 1]
```diff
@@ -956,69 +961,211 @@ func (dm *KubeArmorDaemon) GetSecurityPolicies(identities []string) []tp.Securit
...
 				}
 			}
+		} else if secPolicyType == KubeArmorClusterPolicy {
+			// additional OR check added with containsPolicy() is when this endPoint's ns is removed from secPolicy.Spec.Selector.MatchExpressions[i].Values
+			// due to which secPolicy.Spec.Selector.NamespaceList will not have the removed ns
+			if kl.ContainsElement(secPolicy.Spec.Selector.NamespaceList, endPoint.NamespaceName) || containsPolicy(endPoint.SecurityPolicies, secPolicy) {
+				if action == "ADDED" {
+					// add a new security policy if it doesn't exist
+					new := true
+					for _, policy := range endPoint.SecurityPolicies {
```
Comment (Aryan-sharma11) [inline KubeArmor/core/kubeUpdate.go:1073]: Do we need this check here? Do we have field "namespaceName" in cluster policy?

File: KubeArmor/core/kubeUpdate.go [hunk 2]
```diff
@@ -956,69 +961,211 @@ func (dm *KubeArmorDaemon) GetSecurityPolicies(identities []string) []tp.Securit
...
-					if policy.Metadata["namespaceName"] == secPolicy.Metadata["namespaceName"] && policy.Metadata["policyName"] == secPolicy.Metadata["policyName"] {
-						new = false
-						break
+		if secPolicyType == KubeArmorPolicy {
+			if kl.MatchIdentities(secPolicy.Spec.Selector.Identities, endPoint.Identities) && (len(secPolicy.Spec.Selector.Containers) == 0 || kl.ContainsElement(secPolicy.Spec.Selector.Containers, endPoint.ContainerName)) {
+				if action == "ADDED" {
+					// add a new security policy if it doesn't exist
+					new := true
+					for _, policy := range endPoint.SecurityPolicies {
+						if policy.Metadata["namespaceName"] == secPolicy.Metadata["namespaceName"] && policy.Metadata["policyName"] == secPolicy.Metadata["policyName"] {
```
Comment (Aryan-sharma11) [inline KubeArmor/core/kubeUpdate.go:1029]: can we use `containsPolicy()` here too? `containsPolicy(policy type , endPoint.SecurityPolicies, secPolicy)`.

File: KubeArmor/core/kubeUpdate.go (hunk 1 above)
Comment (Prateeknandle) [reply KubeArmor/core/kubeUpdate.go:1073]: No, we do not need this for cluster policies, will remove this check

Comment (Prateeknandle) [reply KubeArmor/core/kubeUpdate.go:1030]: @daemon1024 should we remove `namespacename` from metadata?

File: KubeArmor/core/kubeUpdate.go (hunk 2 above)
Comment (Prateeknandle) [reply KubeArmor/core/kubeUpdate.go:1029]: no, for `KubeArmorPolicy` the defined two checks are enough. If either of them is not matched, `namespaceName` or `policyName`, then it is a different policy and we need to add a new policy in `endPoint.SecurityPolicies`

File: tests/k8s_env/csp/csp_test.go [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+})
+
+func getNginxPod(name, namespace, ant string) string {
+	pods, err := K8sGetPods(name, namespace, []string{ant}, 60)
+	Expect(err).To(BeNil())
+	Expect(len(pods)).To(Equal(1))
+	return pods[0]
+}
+
+var _ = Describe("Smoke", func() {
```
Comment (daemon1024) [inline tests/k8s_env/csp/csp_test.go]: Rename Test Suit

**Review by daemon1024 — APPROVED** [review]
Tried it out locally, Working as expected!
LGTM, Great Work👌🏽

**Review by Aryan-sharma11 — APPROVED** [review]

**Pull request merged**